library/general/disallowedtags/template.yaml

```diff
@@ -0,0 +1,45 @@
+apiVersion: templates.gatekeeper.sh/v1beta1
+kind: ConstraintTemplate
+metadata:
+  name: k8sdisallowedtags
+  annotations:
+    description: Requires container images to have an image tag different
+      from the ones in a specified list.
+spec:
+  crd:
+    spec:
+      names:
+        kind: K8sDisallowedTags
+      validation:
+        # Schema for the `parameters` field
+        openAPIV3Schema:
+          properties:
+            tags:
+              type: array
+              items:
+                type: string
+  targets:
+    - target: admission.k8s.gatekeeper.sh
+      rego: |
+        package k8sdisallowedtags
+
+        violation[{"msg": msg}] {
+          container := input_containers[_]
+          tags := [forbid | tag = input.parameters.tags[_] ; forbid = endswith(container.image, tag)]
```

Comment:
Should we create an object that maps disallowed tags so that we don't need to scroll through the list every time when there are multiple containers?

@tsandall is there some way to make sure that a map of tags would be created at the entry point of evaluating the rule so we guarantee we aren't paying the cost of construction * the # of images?

In practice, this may be fine as the # of images and the # of disallowed tags may not be very large.